### PR TITLE
Remove block open logic from DBS3Reader and WorkQueue

### LINKED ISSUE #11120: Remove legacy concept of open blocks from DBS
**Impact of the new feature**
WMCore in general (and likely T0 and CRAB)

**Is your feature request related to a problem? Please describe.**
Back in the DBS2 era, we used to have open blocks available in the DBS Server and files were added to them until a given point (until that block was closed). With the DBS3 service, that concept was removed and we can only inject closed blocks in the server.

It turns out that this feature is still supported in WMCore, which makes the Global WorkQueue logic unnecessarily complex. 

**Describe the solution you'd like**
I suggest we remove this functionality from WMCore, which affects (at least):
* DBS3Reader, which should no longer check for open blocks and we should remove the `open_for_writing|OpenForWriting` attributes
* (Global) WorkQueue, removing the same OpenForWriting logic; which affects `OpenForNewData`

Regarding `OpenRunningTimeout`, it's being repurposed with this GH issue: https://github.com/dmwm/WMCore/issues/11048, such that we use this attribute to decide whether we need to look up for new blocks in the input dataset (instead of looking for new files in an open block).

**Describe alternatives you've considered**
This change is not really necessary, but code becomes cleaner and more maintainable without it.

**Additional context**
For now, the 3rd and 4th commit in this PR https://github.com/dmwm/WMCore/pull/11119 contains most of the necessary changes

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #11120
Partial fix for #11048 

#### Status
not-tested

#### Description
As described in the issue, this feature was needed back in the days of DBS2.

This PR removes DBS/WorkQueue logic related to:
* DBS blocks that could be open (onlyClosedBlocks parameter, or open_for_writing, or OpenForWriting)
* DBS3Reader APIs will no longer return a key/value pair for `IsOpen`
* DBS3Reader method `blockIsOpen` has been removed
* DBS3Reader method `listOpenFileBlocks` has been removed (use `listFileBlocks` instead)
* (unused?) DBSWriterObjects function `getDBSFileBlock` has been removed
* WorkQueue Block start policy now defaults `OpenForNewData` to False
* WorkQueue method `addNewFilesToOpenSubscriptions` has been removed, since we no longer add new files to open subscriptions (To be tested: what happens when new elements are pulled down?)
* updated emulators code

#### Is it backward compatible (if not, which system it affects?)
NO (in the sense that it removes a feature)

#### Related PRs
PR https://github.com/dmwm/WMCore/pull/11119 will have to be rebased once this one goes through.

#### External dependencies / deployment changes
None

### CODE REVIEW

**Comment by amaltaro** [comment]
FYI @belforte Stefano, I removed another DBS3Reader method that wasn't mentioned in the initial GH issue. Please check out the first description of this PR with a summary to make sure it won't catch you by surprise ;)

**Comment by belforte** [comment]
thanks @amaltaro . No problem for CRAB, none of those was used.

File: src/python/WMCore/Services/DBS/DBSWriterObjects.py [hunk 1]
```diff
@@ -378,76 +378,4 @@ def createDBSFileBlock(blockName):
...
-#            Name = newBlockName,
-#            Dataset = procDataset,
-#            PhEDExNodeList = [ pnn ]
-#            )
-
-
-
-    logging.warning("Open FileBlock located at PNN: %s to use is FileBlock: %s ", pnn, blockRef['Name'])
-    return blockRef
+    return DbsFileBlock( Name = blockName)
```
Comment (todor-ivanov) [inline src/python/WMCore/Services/DBS/DBSWriterObjects.py:381]: Please, add the new line the end of the file :)

**Review by todor-ivanov — APPROVED** [review]
Thanks @amaltaro 
Looking good to me. Please give us further info upon testing.

**Comment by amaltaro** [comment]
Thanks for the review over the weekend, Stefano and Todor.
Now that we merge it, I can resume my work on https://github.com/dmwm/WMCore/pull/11119

**Pull request merged**